### PR TITLE
Export/335 implement exported type metadata provider

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
+        private static ExportTypePropertyInfo[] GetFromType(Type t, string baseMemberName)
+        {
+            var result = new List<ExportTypePropertyInfo>();
+            foreach (var pi in t.GetProperties())
+            {
+                if (IsNested(pi.PropertyType))
+                {
+                    result.AddRange(GetFromType(pi.PropertyType, pi.Name));
+                }
+                else
```
Comment (yecli) [inline Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs]: I think here we need to check that property is public and have public getter

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/IPagedDataSource.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.ExportModule.Core.Model
+{
+    public interface IPagedDataSource
+    {
+        int PageSize { get; set; }
+        int CurrentPageNumber { get; set; }
+        int GetTotalCount();
+        IEnumerable<Entity> FetchNextPage();
```
Comment (yecli) [inline Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/IPagedDataSource.cs]: Not sure we should restrict IEnumerable to Entities here. Lets leave non generic IEnumerable, as IExportProvider could write any object.

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs (hunk 1 above)
Comment (akak1977) [reply Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs]: GetProperties в этой реализации делает это по-умолчанию.

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/IPagedDataSource.cs (hunk 2 above)
Comment (akak1977) [reply Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/IPagedDataSource.cs]: Исправил

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs (hunk 1 above)
Comment (yecli) [reply Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs]: Also here we in fact need to implement IEnumerable check, as dependencies are usually stored inside IEnumerable or array (which is IEnumerable too in runtime), and to navigate then by Generic type.
Also, need to add unit tests for this for some objects hierarchy with collections, array.

**Pull request merged**